test/unit/repositories/test_catalogue_category.py

```diff
@@ -27,1334 +37,1204 @@
-
-    retrieved_breadcrumbs = catalogue_category_repository.get_breadcrumbs(catalogue_category_id)
-
-    mock_utils.create_breadcrumbs_aggregation_pipeline.assert_called_once_with(
-        entity_id=catalogue_category_id, collection_name="catalogue_categories"
-    )
-    mock_utils.compute_breadcrumbs.assert_called_once_with(
-        list(MOCK_BREADCRUMBS_QUERY_RESULT_LESS_THAN_MAX_LENGTH),
-        entity_id=catalogue_category_id,
-        collection_name="catalogue_categories",
-    )
-    assert retrieved_breadcrumbs == mock_breadcrumbs
-
-
-def test_list(test_helpers, database_mock, catalogue_category_repository):
-    """
-    Test getting catalogue categories.
-
-    Verify that the `list` method properly handles the retrieval of catalogue categories without filters.
-    """
+    def mock_breadcrumbs(self, breadcrumbs_query_result: list[dict]):
+        """Mocks database methods appropriately to test the 'get_breadcrumbs' repo method
+
+        :param breadcrumbs_query_result: List of dictionaries containing the breadcrumbs query result from the
+                                         aggregation pipeline
+        """
+
+        self._breadcrumbs_query_result = breadcrumbs_query_result
+        self._mock_aggregation_pipeline = MagicMock()
+        self._expected_breadcrumbs = MagicMock()
+
+        self.mock_utils.create_breadcrumbs_aggregation_pipeline.return_value = self._mock_aggregation_pipeline
+        self.catalogue_categories_collection.aggregate.return_value = breadcrumbs_query_result
+        self.mock_utils.compute_breadcrumbs.return_value = self._expected_breadcrumbs
+
+    def call_get_breadcrumbs(self, catalogue_category_id: str):
+        """Calls the CatalogueCategoryRepo `get_breadcrumbs` method"""
+
+        self._obtained_catalogue_category_id = catalogue_category_id
+        self._obtained_breadcrumbs = self.catalogue_category_repository.get_breadcrumbs(
+            catalogue_category_id, session=self.mock_session
+        )
+
+    def check_get_breadcrumbs_success(self):
+        """Checks that a prior call to `call_get_breadcrumbs` worked as expected"""
+
+        self.mock_utils.create_breadcrumbs_aggregation_pipeline.assert_called_once_with(
+            entity_id=self._obtained_catalogue_category_id, collection_name="catalogue_categories"
+        )
+        self.catalogue_categories_collection.aggregate.assert_called_once_with(
+            self._mock_aggregation_pipeline, session=self.mock_session
+        )
+        self.mock_utils.compute_breadcrumbs.assert_called_once_with(
+            list(self._breadcrumbs_query_result),
+            entity_id=self._obtained_catalogue_category_id,
+            collection_name="catalogue_categories",
+        )
+
+        assert self._obtained_breadcrumbs == self._expected_breadcrumbs
+
+
+# pylint: disable=duplicate-code
+class TestGetBreadcrumbs(GetBreadcrumbsDSL):
+    """Tests for getting the breadcrumbs of a Catalogue Category"""
```

Comment:
```suggestion
    """Tests for getting the breadcrumbs of a Catalogue Category"""
```
When I did systems I used uppercase as I was thinking of them as a specific entity, like a name, so I did this here for consistency, but I am not sure on it let me know if you have  a preference @VKTB. Its also a source of inconsistency in some of my own error messages e.g. `Cannot move a system to one of its own children` is different to the rest, I will change either that or the others as well.

Reply:
I didn't think of them as entities which is why I wrote them the other way. It becomes a little trickier with the plural form if you see them as entities.

Reply:
I will change Systems to systems to be consistent.